FixDevV3Blobs/FixDevV3Blobs/Program.cs

```diff
@@ -0,0 +1,279 @@
+            }
+        }
+
+        private static async Task<ConcurrentBag<CloudBlockBlob>> LoadBlobListAsync(CloudBlobContainer container)
+        {
+            var blobList = new ConcurrentBag<CloudBlockBlob>();
+
+            BlobContinuationToken continuationToken = null;
+            var stopwatch = Stopwatch.StartNew();
+            do
+            {
+                var segment = await container.ListBlobsSegmentedAsync(
+                    prefix: null,
+                    useFlatBlobListing: true,
+                    blobListingDetails: BlobListingDetails.None,
+                    maxResults: 5000,
+                    currentToken: continuationToken,
+                    options: null,
+                    operationContext: null);
+
+                var range = segment.Results.OfType<CloudBlockBlob>().ToList();
+                if (range.Count != segment.Results.Count())
+                {
+                    throw new InvalidOperationException("Unexpected result item type: " + segment.Results.First(r => r.GetType() != typeof(CloudBlockBlob)).GetType());
+                }
+                range.ForEach(b => blobList.Add(b));
+                continuationToken = segment.ContinuationToken;
+                Log($"Discovered {blobList.Count} blobs in {stopwatch.Elapsed}");
+            } while (continuationToken != null);
+            stopwatch.Stop();
+            return blobList;
+        }
+
+        private static async Task<Stopwatch> ProcessBlobListAsync(ConcurrentBag<CloudBlockBlob> blobList, string search, string replace)
+        {
+            var stopwatch = Stopwatch.StartNew();
+            var count = 0;
+            var skipped = 0;
+            var blobCount = blobList.Count;
+
+            var tasks = Enumerable.Range(1, MaxTasks).Select(async _ => 
+            {
+                await Task.Yield();
+                while (blobList.TryTake(out var blob))
+                {
+                    var curCount = Interlocked.Increment(ref count);
+                    if (Processed(blob))
+                    {
+                        Interlocked.Increment(ref skipped);
+                        continue;
+                    }
+                    int attempt = 0;
+                    do
+                    {
+                        try
+                        {
+                            await ProcessBlobAsync(blob, search, replace);
+                            break;
+                        }
+                        catch (UnsupportedContentEncodingException e)
+                        {
+                            Log($"{blob.Uri.AbsoluteUri}: {e.Message}");
+                            attempt = MaxRetries;
+                        }
+                        catch (Exception e)
+                        {
+                            Log($"Got exception while processing {blob.Uri.AbsoluteUri}: {e}");
+                            await Task.Delay(TimeSpan.FromSeconds(10));
```

Comment:
If exception will the blob need to be added back to the bag?

Reply:
The tool tracks successfully processed jobs and if started again it skips those. So no retrying is done beyond something very simple. It is assumed that whoever runs the tool will rerun it if there were exceptions.